scripts: Typo in the word 'platform'

LukaszMrugala opened the pull request:
`testinstance.py` contains a string literal comparison to the word `platform`, which has been misspelled as `plaform`.

That typo and a few others, contained within docstrings, have been corrected to `platform`.

The change is verified by a test expansion.

Fixes #58322

golowanow commented on scripts/tests/twister/test_testinstance.py:
```diff
@@ -56,22 +56,30 @@ def test_check_build_or_run(class_testplan, monkeypatch, all_testsuites_dict, pl
     testcase = class_testplan.testsuites.get('scripts/tests/twister/test_data/testsuites/samples/test_app/sample_test.app')
+
+    if extra_configs:
+        testcase.extra_configs = extra_configs
```
@LukaszMrugala please split the `extra_configs` fix and the typo fix into separate commits and update the PR's description that there are two fixes here.

LukaszMrugala commented:
`CONFIG_NULL` changed to `CONFIG_LOG`. `CONFIG_NULL` is not defined anywhere, so it failed compliance checks. `CONFIG_LOG` has been deemed as close to a `null` config (one that does nothing), so it is used instead.

nashif commented:
Can you please split the typo fixes in various tests into their own commits?

LukaszMrugala commented:
> Can you please split the typo fixes in various tests into their own commits?

Yes, I could do that. I am assuming that you'd like one commit for `test_*.py` and `testinstance.py` change and then one commit for every `*.c` typo changed?

nashif commented:
> Yes, I could do that. I am assuming that you'd like one commit for `test_*.py` and `testinstance.py` change and then one commit for every `*.c` typo changed?

no, one commit for the twister pytest tests, another for tests/ztest/.. and one more for test/lib/...

LukaszMrugala committed:
scripts: Typo in the word 'platform' fix

testinstance.py contains a string literal comparison to the
word 'platform', which has been misspelled as 'plaform'.

The change is verified by a test expansion.

Signed-off-by: Lukasz Mrugala <lukaszx.mrugala@intel.com>

LukaszMrugala committed:
testsuite: Docstring typo fix

Word 'platform' misspelled as 'platorm'.

Signed-off-by: Lukasz Mrugala <lukaszx.mrugala@intel.com>

LukaszMrugala committed:
tests: shared_multi_heap: Docstring typo fix

Word 'platform' misspelled as 'plaform'.

Signed-off-by: Lukasz Mrugala <lukaszx.mrugala@intel.com>

LukaszMrugala committed:
tests: lib: devicetree: Docstring typo fix

Word 'platform' misspelled as 'plaform'.

Signed-off-by: Lukasz Mrugala <lukaszx.mrugala@intel.com>

golowanow commented:
@LukaszMrugala the two new commits have identical description, could change subjects there like this:
`tests: lib: devicetree: Fixes docstring typo`
`tests: shared_multi_heap: Fixes docstring typo`
